src/main/java/io/confluent/idesidecar/restapi/models/graph/RealDirectFetcher.java

```diff
@@ -112,6 +112,11 @@ public List<DirectConnection> getConnections() {
   public Uni<DirectKafkaCluster> getKafkaCluster(String connectionId) {
     var state = connections.getConnectionState(connectionId);
     if (state instanceof DirectConnectionState directState) {
+      if (!directState.isKafkaConnected()) {
+        // Either there is no Kafka cluster configured or it is not connected, so return no info
+        Log.debugf("Skipping connection '%s' since Kafka is not connected.", connectionId);
+        return Uni.createFrom().nullItem();
+      }
```

Comment:
We might also consider either doing the same logic in other `Real*Fetcher` implementations, or moving this logic into the different GraphQL resources. (Seems better here, but I'm open to other suggestions.)